lib/src/component/abstract_transition.dart

```diff
@@ -123,6 +123,9 @@ abstract class AbstractTransitionComponent<T extends AbstractTransitionProps, S
   /// Whether the Element returned by [getTransitionDomNode] will have a transition event.
   bool get hasTransition => true;
 
+  /// The duration that can elapse before a transition timeout occurs.
+  Duration get transitionTimeout => const Duration(seconds: 15);
```

Comment:
I feel like a getter is better for this, similar to `hasTransition`. That way consumers have the ability to either proxy it with a prop or have a custom implementation.

1s sounds good to me tho.